### DISCUSSION TITLE
how can i disable logging?

### DISCUSSION THREAD

**Discussion by danila-schelkov** [encode/httpx#2733] [Q&A] [answered]
I have a ton of log messages from python-telegram-api and i want to disable logging for this library.
![image](https://github.com/encode/httpx/assets/54549682/44f89bfa-bcbe-458c-a394-fa6306c2523a)

I've tried this: `logging.getLogger("httpx").handlers.clear()`, but it doesn't work

**Comment by zanieb** [comment, marked as answer]
`logging.getLogger("httpx").setLevel("CRITICAL")` should do it — the handler is likely for your "root" logger. You could also disable propagation `logging.getLogger("httpx").propagate = False`

**Reply by danila-schelkov** [reply]
Thank you so much!
